extensions/amp-analytics/0.1/requests.js

```diff
@@ -102,13 +131,25 @@ export class RequestHandler {
       });
     };
 
+    let batchSegmentResolver = null;
+    this.batchSegmentsPromise_.push(new Promise(resolve => {
```

Comment:
instead of creating a separate promise, can you reuse the one below?

Reply:
fixed